src/Phinx/Db/Adapter/PostgresAdapter.php

```diff
@@ -481,7 +481,9 @@ public function getColumns(string $tableName): array
                     self::PHINX_TYPE_BIG_INTEGER,
                 ], true)
             ) {
-                $column->setPrecision($columnInfo['numeric_precision']);
+                if (isset($columnInfo['numeric_precision'])) {
+                    $column->setPrecision($columnInfo['numeric_precision']);
+                }
```

Comment:
Should we maybe rather change the type check above, and explicitly check for the numeric types for which Phinx supports specifying precision? This "**is _not_ of type**" check seems very prone to possible future regressions.

Reply:
I first tried to add STRING and CHAR to the `!in_array` above.  This did work, but, as a neophyte to Phinx, I was concerned I might be missing a type, which echoes your concerns.   

https://www.postgresql.org/docs/current/infoschema-columns.html says numeric_precision contains "the (declared or implicit) precision" for numeric types and null for all others.  I think the implicit precisions is how we got to you `!in_array`. 

Reviewing Phinx to see where `getPrecision()` is used, I see some adapters use it if it contains a numeric value, no matter what type is being used. However, PostgresAdapter only uses it for:
* PHINX_TYPE_DECIMAL
* PHINX_TYPE_TIME
* PHINX_TYPE_TIMESTAMP (`getSqlType` looks to turn a PHINX_TYPE_DATETIME read by getColumns() into an associated array with 'name' => 'timestamp', so this works with the `if (in_array($columnType, [static::PHINX_TYPE_TIME, static::PHINX_TYPE_DATETIME], true)) {`, but it did drive me in a marry little circle. 

So, I think PHINX_TYPE_DECIMAL is the extent of it. Yeah?

The following code does pass locally:

```
if (in_array($columnType, [static::PHINX_TYPE_TIME, static::PHINX_TYPE_DATETIME], true)) {
    $column->setPrecision($columnInfo['datetime_precision']);
} elseif ($columnType === self::PHINX_TYPE_DECIMAL) {
    $column->setPrecision($columnInfo['numeric_precision']);
}
```

Reply:
Yeah, it very much looks like decimal is the only type for which the Postgres adapter supports numerical precision.

Reply:
Do we need any further changes?